benchmarks/matmul_benchmark/implementation.py

```diff
@@ -0,0 +1,50 @@
+from common.benchmark_wrapper import BenchmarkWrapper
+import json
+import os
+import torch
+import time
+
+
+class MatmulBenchmark(BenchmarkWrapper): #it's the same class inside benchmark_info.json
+
+    def __init__(self):
+        self.filePath = os.path.dirname(__file__) #os.path.dirname(__file__) returns the path of this file
+
+    def setSettings(self, settings_file):
+        """
+        Open and read the settings file
+        """
+        settings = os.path.join(self.filePath, "settings", settings_file)
+        self.dimension1 = json.load(open(settings, "r"))["dimension1"]
+        self.dimension2 = json.load(open(settings, "r"))["dimension2"]
+
+    def startBenchmark(self, verbosity=None):
+
+        start_time = time.time()
+
+        print("The first matrix has dimensions", self.dimension1, "x", self.dimension2)
+        print("The second matrix has dimensions", self.dimension2, "x", self.dimension1)
+
+        matrix_1 = torch.randn(self.dimension1, self.dimension2, device="cpu")

```

Comment:
Imho the matrix allocation time should be a different time measurement e.g. one timer for allocation, one timer for actual multiplication. Allocation and multiplication are quite different tasks and measuring them all together might hide some important information.

Reply:
Fixed!